web/src/invite.ts

```diff
@@ -358,6 +357,10 @@ function open_invite_user_modal(e: JQuery.ClickEvent<Document, undefined>): void
             get_text_from_item: email_pill.get_email_from_item,
         });
 
+        const $stream_pill_container = $("#invite_streams_container .pill-container");
+
+        stream_pill_widget = add_streams_pill.create($stream_pill_container);
```

Comment:
So `render_invite_user_modal` is passed a `streams` parameter, and maybe some others, that are now unused; probably needs various bits of code deleted. Notably that started with the set `get_invite_streams(); I'm not sure your new `add_streams_pill` module uses the correct logic?

Reply:
> I'm not sure your new `add_streams_pill` module uses the correct logic?

The function `get_invite_streams()` currently returns all default streams plus the streams the inviting user is subscribed to, sorted alphabetically. However, since we're now using pills, alphabetical order is unnecessary because users can search for streams directly. Additionally, do you want to restrict the search results in the input box? Currently, users can search through all streams, do you want to limit it to only default streams and the streams the inviting user is subscribed to?

Reply:
Hmm, yeah. I think we want to limit to exactly the set of streams that the current user as permission to add another user to. 

But also if you're removing the last caller to `get_invite_streams`, we probably want to delete that function. But it might be just as reasonable to rework the function to use the correct rule, removing the sorting, and still call it.

Reply:
I suggest removing the `get_invite_stream`s function. Its functionality is essentially redundant as it merely sorts the results provided by `stream_data.get_invite_stream_data`, which already performs the necessary work.

We should use and rework `stream_data.get_invite_stream_data` instead.